Documentation/deployment.md

```diff
@@ -5,80 +5,125 @@ This guide walks through deploying the `bootcfg` service on a Linux host (via bi
+`bootcfg` is a service for network booting and provisioning machines to create CoreOS clusters. `bootcfg` should be installed on a provisioner machine (CoreOS or any Linux distribution) or cluster (Kubernetes) which can serve configs to client machines in a lab or datacenter.
 
-`bootcfg` should be deployed to a provisioner node (CoreOS or any Linux distribution) or cluster (Kubernetes) which can serve configurations to client machines in a lab or datacenter. Choose a deployment target and a DNS name (or IP) at which it will be accessed (e.g. bootcfg.example.com).
+Choose one of the supported installation options:
 
-## Binary Release
+* [CoreOS (systemd & rkt)](#coreos)
+* [Generic Linux (systemd & binary)](#binary-release)
+* [With rkt](#rkt)
+* [With docker](#docker)
+* [Kubernetes Service](#kubernetes)
+* RPM package ([coming soon](https://github.com/coreos/coreos-baremetal/issues/266))
+
+## Download
 
 Download the latest coreos-baremetal [release](https://github.com/coreos/coreos-baremetal/releases) to the provisioner host.
 
-    wget https://github.com/coreos/coreos-baremetal/releases/download/v0.4.0/coreos-baremetal-v0.4.0-linux-amd64.tar.gz
-    wget https://github.com/coreos/coreos-baremetal/releases/download/v0.4.0/coreos-baremetal-v0.4.0-linux-amd64.tar.gz.asc
+```sh
+$ wget https://github.com/coreos/coreos-baremetal/releases/download/v0.4.0/coreos-baremetal-v0.4.0-linux-amd64.tar.gz
+$ wget https://github.com/coreos/coreos-baremetal/releases/download/v0.4.0/coreos-baremetal-v0.4.0-linux-amd64.tar.gz.asc
+```
 
 Verify the release has been signed by the [CoreOS App Signing Key](https://coreos.com/security/app-signing-key/).
 
-    gpg --keyserver pgp.mit.edu --recv-key 18AD5014C99EF7E3BA5F6CE950BDD3E0FC8A365E
-    gpg --verify coreos-baremetal-v0.4.0-linux-amd64.tar.gz.asc coreos-baremetal-v0.4.0-linux-amd64.tar.gz
-    # gpg: Good signature from "CoreOS Application Signing Key <security@coreos.com>"
+```
+$ gpg --keyserver pgp.mit.edu --recv-key 18AD5014C99EF7E3BA5F6CE950BDD3E0FC8A365E
+$ gpg --verify coreos-baremetal-v0.4.0-linux-amd64.tar.gz.asc coreos-baremetal-v0.4.0-linux-amd64.tar.gz
+# gpg: Good signature from "CoreOS Application Signing Key <security@coreos.com>"
+```
 
 Untar the release.
 
-    tar xzvf coreos-baremetal-v0.4.0-linux-amd64.tar.gz
-    cd coreos-baremetal-v0.4.0-linux-amd64
+```sh
+$ tar xzvf coreos-baremetal-v0.4.0-linux-amd64.tar.gz
+$ cd coreos-baremetal-v0.4.0-linux-amd64
+```
+
+## TLS Credentials
 
-### Copy Binary
+*Skip this unless you need to enable the gRPC API*
+
+The `bootcfg` gRPC API allows external client tools (`bootcmd` CLI, graphical apps) to manage how machines are provisioned so TLS credentials are needed for client authentication and to establish a secure communication channel. Client machines read from the HTTP endpoints and do not require this setup.
```

Comment:
"Client machines" here is a bit vague. Maybe be more explicit as to what type of clients. 

i.e. "Client machines which are PXE booting"